File: model-client/src/main/java/org/modelix/model/lazy/IDeserializingKeyValueStore_extensions.java
package org.modelix.model.lazy;



import org.modelix.model.persistent.HashUtil;
import org.modelix.model.persistent.CPElement;

public class IDeserializingKeyValueStore_extensions {
  public IDeserializingKeyValueStore_extensions() {
  }
  public static void put(final IDeserializingKeyValueStore _this, Object deserialized, String serialized) {
    _this.put(HashUtil.sha256(serialized), deserialized, serialized);
  }
  public static void put(final IDeserializingKeyValueStore _this, CPElement element) {
    IDeserializingKeyValueStore_extensions.put(_this, element, element.serialize());
  }
}

File: model-client/src/main/kotlin/org/modelix/model/lazy/IDeserializingKeyValueStore_extensions.kt
package org.modelix.model.lazy

import org.modelix.model.persistent.CPElement
import org.modelix.model.persistent.HashUtil

object IDeserializingKeyValueStore_extensions {
    fun put(_this: IDeserializingKeyValueStore, deserialized: Any?, serialized: String?) {
        _this.put(HashUtil.sha256(serialized), deserialized, serialized)
    }

    @JvmStatic
    fun put(_this: IDeserializingKeyValueStore, element: CPElement) {
        put(_this, element, element.serialize())
    }
}